### ISSUE TITLE
[CWM] Sort search results alphabetically

### ISSUE THREAD

**Issue by ColmDC** [DigitalCommons/mykomap-monolith#89]
**Description**
There is currently no descernable order to the search results.
Sort them alphabetically

![Image](https://github.com/user-attachments/assets/161d530d-4e39-4115-af5b-c9dd367ed6cf)

**Acceptance Criteria**
All search results appear in alphabetical order.

**Comment by rogup** [comment]
This isn't trivial, so will have to wait until after Delhi.

Alphabetic sorting by name needs to happen before or during the generation of the BE data. We can't sort on the front-end since data is paginated and we don't have the full set of names. The server can't do it when loading the data, since sorting data would change the order of indexes, which then wouldn't match the item data files. And we can't do a sort before we search every set of search results, since this would take a lot of computation power and impact performance.

I think the best time to do the sorting is as part of the CSV generation, which is @wu-lee's realm of knowledge. Maybe just with a simple csv sort command line script.

**Comment by ColmDC** [comment]
Grand. Moved into backlog.

**Comment by ColmDC** [comment]
@wu-lee Can you comment on how much effort this might take. Might be good time and alinged with walk through with @King-Mob?

**Comment by wu-lee** [comment]
The conversion step from CSV -> MM4.x datastructures would be doing this bit. I would say it's a feature request for the code which executes that step, rather than a an aspect of the overall design, but it can certainly be prioritised to come out of the work at some point.

This will hopefully be clearer when we publish some diagrams.

**Comment by rogup** [comment]
@wu-lee Do you mean this will be part of the work that @King-Mob is working on?

Or is it ok for me to go ahead with this ticket and add the alphabetisation to the database-cli script that converts csv -> MM4.X folder structure?

**Comment by wu-lee** [comment]
No, I mean exactly the database-cli script is the thing which does this, and not the merge/normalisation stuff.

I was thinking of maybe doing this next, but sure, if you want to do this that's fine, maybe I can try and do the bash-to-typescript spike.

**Comment by rogup** [comment]
@wu-lee You can take it. I've moved on to another ticket

**Comment by wu-lee** [comment]
> Is anyone actively working on this?
I started, but then switched to reviewing some of the PRs in the sprint, so others weren't blocked.

Revisiting: my conclusion, having had a look at the code and a think, is that there's no need to sort the MM data in the CLI. This is because the order of the data in the CSV file is preserved when importing. Therefore, all we need to do is sort the CSV first, before importing it. No code changes needed.

This has the small added benefit that we can sort datasets however we like, and it isn't hardwired to be sorted, say, alphabetically by name.

I've checked this is true locally. However I think the right time to redeploy the sorted version of data on the server is when we have the new Q3 dataset ready, as there's a bit of urgency and this reduces the number of redeploys.

Also: I did notice what seems to be a bug in the pagination of the sort results. Instead of code for the above, I'll submit a fix for that in the associated PR.
